File: ComputerSeekho/src/main/java/com/example/entities/Staff.java
package com.example.entities;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.*;
import jakarta.validation.constraints.*;
import lombok.Data;

@Entity
@Data
@Table(name = "staff")
public class Staff {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "staff_id")
    private int staffId;

    @Column(name = "staff_name")
    @NotBlank
    private String staffName;

    @Column(name = "photo_url")
    @NotBlank
    private String photoUrl;

    @Column(name = "staff_email")
    @NotBlank
    @Email
    private String staffEmail;

    @Column(name = "staff_mobile")
    @NotBlank
    private String staffMobile;

    @Column(name = "staff_gender")
    @NotBlank
    private String staffGender;

    @Column(name = "staff_username")
    @NotBlank
    @Size(min = 5, max = 30)
    private String staffUsername;

    @Column(name = "staff_password")
    @NotBlank
    @Size(min = 8)
    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
    private String staffPassword;

    @Column(name = "staff_role")
    private String staffRole;
}

File: ComputerSeekho.Net/ComputerSeekho.Net/Models/Staff.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace ComputerSeekho.Net.Models;

[Table("staff")]
public class Staff
{
    [Key]
    [Column("staff_id")]
    public int StaffId { get; set; }

    [Column("staff_name")]
    [Required]
    public string StaffName { get; set; } = string.Empty;

    [Column("photo_url")]
    [Required]
    public string PhotoUrl { get; set; } = string.Empty;

    [Column("staff_email")]
    [Required]
    [EmailAddress]
    public string StaffEmail { get; set; } = string.Empty;

    [Column("staff_mobile")]
    [Required]
    public string StaffMobile { get; set; } = string.Empty;

    [Column("staff_gender")]
    [Required]
    public string StaffGender { get; set; } = string.Empty;

    [Column("staff_username")]
    [Required]
    [MaxLength(30)]
    public string StaffUsername { get; set; } = string.Empty;

    [Column("staff_password")]
    [Required]
    [MinLength(8)]
    // JsonIgnore for WriteOnly equivalent in separate DTOs usually, but for Entity direct use we can suppress serialization
    // [JsonIgnore] // Use DTOs for returning data to avoid sending password. Java used @JsonProperty(access = WRITE_ONLY)
    public string StaffPassword { get; set; } = string.Empty;

    [Column("staff_role")]
    public string? StaffRole { get; set; }
}
